src/parser/mod.rs

```diff
@@ -2266,6 +2312,109 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse an internal data type of clickhouse nullable [1]
+    /// Internal data type of nullable can’t be any of the composite data types Array, Map and Tuple
+    /// Syntax
+    /// ```sql
+    /// Nullable(data_type)
+    /// ```
+    ///
+    /// [1]: https://clickhouse.com/docs/en/sql-reference/data-types/nullable
+    fn parse_click_house_nullable(&mut self) -> Result<DataType, ParserError> {
+        self.expect_keyword(Keyword::NULLABLE)?;
+
+        self.expect_token(&Token::LParen)?;
+        let (inside_type, _) = self.parse_data_type_helper()?;
+        if matches!(
+            inside_type,
+            DataType::Array(_) | DataType::Map(_, _) | DataType::Tuple(_)
+        ) {
+            return parser_err!("Unexpected internal data type of nullable:", inside_type);
+        }
```

Comment:
I think we can skip this check as it falls towards the semantics side of things - if we have a nested type we can always embedded it within the nullable

Reply:
Makes sense. I removed this condition.